packages/shared/src/components/auth/AuthOptions.tsx

```diff
@@ -239,70 +238,67 @@ function AuthOptions({
     onSetActiveDisplay(AuthDisplay.CodeVerification);
   };
 
-  useWindowEvents<SocialRegistrationFlow>(
-    'message',
-    AuthEvent.SocialRegistration,
-    async (e) => {
-      if (e.data?.flow) {
-        const connected = await getKratosFlow(
-          AuthFlow.Registration,
-          e.data.flow,
-        );
+  useEventListener(globalThis, 'message', async (e) => {
+    if (e.data?.eventKey !== AuthEvent.SocialRegistration) {
```

Comment:
Can this condition be added directly to the `useEventListener`? E.g. by passing another parameter to the `useEventListener` hook. I see that we have the same behavior everywhere where the hook is used (returning undefined), just the condition is different...

Reply:
This was what the old hook did and was hidden with `validateKey`. I think this (while adding some code) at least keeps the condition visible when looking at the related code. And we don't really use it on much of the places.

Reply:
I could add the new hook like `useKratosMessageListener` that would encapsulate that logic and subscribe to specific key?